src/vl-map-search.js

```diff
@@ -1,108 +1,121 @@
-                            if (data && data.SuggestionResult) {
-                                const resultaten = data.SuggestionResult.map((resultaat) => {
-                                    return {
-                                        value: resultaat,
-                                        label: resultaat
-                                    }
-                                });
-                                this._selectElement.choices = resultaten;
-                            }
-                        });
+  _addSearchEventListener() {
+    if (!this.__searchEventListenerRegistered) {
+      this.__searchEventListenerRegistered = true;
+      this._selectElement.addEventListener('search', (event) => {
+        if (event && event.detail && event.detail.value) {
+          fetch(this.searchUrl + event.detail.value)
+          .then((response) => {
+            return response.json();
+          }).then((data) => {
+            if (data && data.SuggestionResult) {
+              const resultaten = data.SuggestionResult.map((resultaat) => {
+                return {
+                  value: resultaat,
+                  label: resultaat
                 }
-            });
+              });
+              this._selectElement.choices = resultaten;
+            }
+          });
         }
+      });
     }
+  }
 
-    _addChoiceEventListener() {
-        if (!this.__choiceEventListenerRegistered) {
-            this.__choiceEventListenerRegistered = true;
-            this.__choiceEventListener = this._selectElement.addEventListener('choice', (event) => {
-                if (event && event.detail && event.detail.choice) {
-                    fetch(this.locationUrl + event.detail.choice.value)
-                        .then((response) => {
-                            return response.json();
-                        }).then((data) => {
-                            if (data && data.LocationResult) {
-                                this._map.zoomTo([data.LocationResult[0].BoundingBox.LowerLeft.X_Lambert72, data.LocationResult[0].BoundingBox.LowerLeft.Y_Lambert72, data.LocationResult[0].BoundingBox.UpperRight.X_Lambert72, data.LocationResult[0].BoundingBox.UpperRight.Y_Lambert72]);
-                            }
-                        });
+  _addChoiceEventListener() {
+    if (!this.__choiceEventListenerRegistered) {
+      this.__choiceEventListenerRegistered = true;
+      this.__choiceEventListener = this._selectElement.addEventListener(
+          'choice', (event) => {
+            if (event && event.detail && event.detail.choice) {
+              fetch(this.locationUrl + event.detail.choice.value)
+              .then((response) => {
+                return response.json();
+              }).then((data) => {
+                if (data && data.LocationResult) {
+                  if (this._onSelect) {
+                    this._onSelect(data);
+                  }
+                  this._map.zoomTo(
```

Comment:
Door introductie van het binden van de vl-map-search en vl-map is het niet zeker dan this._map altijd bestaat. Ik heb daarom hier wat defensieve code toegevoegd om undefined issues te voorkomen.

Reply:
> Door introductie van het binden van de vl-map-search en vl-map is het niet zeker dan this._map altijd bestaat. Ik heb daarom hier wat defensieve code toegevoegd om undefined issues te voorkomen.

Vl-map-search moet toch altijd met een vl-map gebonden is? Ik zou hier ten minstens een error log toevoegen, anders failed dit "silently".

Reply:
@ZhaodiFengVlaanderen wanneer je als afnemer besloten hebt om zelf de binding te doen tussen vl-map en vl-map-search weet de afnemer dat zolang hij de binding zelf niet doet er geen interactie mogelijk is. Het zou bijvoorbeeld kunnen dat de afnemer eerst de gebruiker iets wilt laten doen met de vl-map-search maar nog geen interactie wilt met de vl-map. In dat geval is het ontbreken van de interactie gewenst gedrag. Daarom heb ik besloten om de undefined issues op te lossen en geen error te gooien. Een info of warning tonen kan natuurlijk informatief wel interessant zijn. Aarzel zeker niet om dit toe te voegen.